### PR TITLE
[JSC] Add forceUnlinkedDFG option

### PR DESCRIPTION
**Pull request by Constellation**
#### 96f919b9cc1050fd14cb0eb16856033abd4d8de5
<pre>
[JSC] Add forceUnlinkedDFG option
<a href="https://bugs.webkit.org/show_bug.cgi?id=239751">https://bugs.webkit.org/show_bug.cgi?id=239751</a>

Reviewed by Saam Barati.

This patch adds forceUnlinkedDFG option to start running tests with unlinked DFG (while it is not truly unlinked yet).

* Tools/Scripts/run-jsc-stress-tests:
* Source/JavaScriptCore/jit/JITOperations.cpp:
(JSC::JSC_DEFINE_JIT_OPERATION):
* Source/JavaScriptCore/runtime/OptionsList.h:
* JSTests/stress/arith-abs-on-various-types.js:
* JSTests/stress/arith-abs-to-arith-negate-range-optimizaton.js:
* JSTests/stress/arith-acos-on-various-types.js:
* JSTests/stress/arith-acosh-on-various-types.js:
* JSTests/stress/arith-asin-on-various-types.js:
* JSTests/stress/arith-asinh-on-various-types.js:
* JSTests/stress/arith-atan-on-various-types.js:
* JSTests/stress/arith-atanh-on-various-types.js:
* JSTests/stress/arith-cbrt-on-various-types.js:
* JSTests/stress/arith-ceil-on-various-types.js:
* JSTests/stress/arith-clz32-on-various-types.js:
* JSTests/stress/arith-cos-on-various-types.js:
* JSTests/stress/arith-cosh-on-various-types.js:
* JSTests/stress/arith-expm1-on-various-types.js:
* JSTests/stress/arith-floor-on-various-types.js:
* JSTests/stress/arith-fround-on-various-types.js:
* JSTests/stress/arith-log-on-various-types.js:
* JSTests/stress/arith-log10-on-various-types.js:
* JSTests/stress/arith-log2-on-various-types.js:
* JSTests/stress/arith-round-on-various-types.js:
* JSTests/stress/arith-sin-on-various-types.js:
* JSTests/stress/arith-sinh-on-various-types.js:
* JSTests/stress/arith-sqrt-on-various-types.js:
* JSTests/stress/arith-tan-on-various-types.js:
* JSTests/stress/arith-tanh-on-various-types.js:
* JSTests/stress/arith-trunc-on-various-types.js:
* JSTests/stress/fold-multi-get-by-offset-to-get-by-offset-with-watchpoint.js:

Canonical link: <a href="https://commits.webkit.org/250009@main">https://commits.webkit.org/250009@main</a>
git-svn-id: <a href="https://svn.webkit.org/repository/webkit/trunk@293473">https://svn.webkit.org/repository/webkit/trunk@293473</a> 268f45cc-cd09-0410-ab3c-d52691b4dbfc
</pre>

### CODE REVIEW

File: Tools/Scripts/run-jsc-stress-tests [hunk 1]
```diff
@@ -1118,6 +1118,7 @@ BASE_MODES = [
             "--useSamplingProfiler=true",
             "--airForceIRCAllocator=true",
             "--useDataICInOptimizingJIT=true",
+            "--forceUnlinkedDFG=true",
```
Comment (saambarati) [inline Tools/Scripts/run-jsc-stress-tests]: Do we really want to start running tests? Might be limiting in future patches that don't fix all issues

**Review by saambarati — APPROVED** [review]

File: Tools/Scripts/run-jsc-stress-tests (hunk 1 above)
Comment (Constellation) [reply Tools/Scripts/run-jsc-stress-tests]: I would like to keep things working while landing a patch so I expect that this option is OK. And we can disable it anytime if we think that's better.

**Comment by Constellation** [comment]
Merge queue looks not working, I'll land it directly.

**Comment by Constellation** [comment]
Landed https://commits.webkit.org/250009@main (96f919b9cc10)!

**Pull request merged**